### PR TITLE
Install missing global

### PR DESCRIPTION
**Pull request by anijain2305**
Related to https://github.com/pytorch/torchdynamo/issues/1126

Not really sure if this is the right fix. My understanding is the `output.root_globals` is the final `f_globals` passed on to the modified frame. During inlining, we need to ensure that `globals` from the inlined functions/files are also reflected in the `output.root_globals`. This fixes the speech_transformer issue.

@voznesenskym @jansel

### CODE REVIEW

File: torchdynamo/symbolic_convert.py [hunk 1]
```diff
@@ -399,6 +399,8 @@ def get_global_source(self, name):
                 source = AttrSource(
                     self.import_source(self.f_globals["__name__"]), name
                 )
+                if name not in self.output.root_globals and name in self.f_globals:
+                    self.output.install_global(name, self.f_globals[name])
```
Comment (voznesenskym) [inline torchdynamo/symbolic_convert.py:403]: is this safe for all globals? Should we check if the value is a module?

**Review by voznesenskym — APPROVED** [review]

**Comment by jansel** [comment]
Hrm... this seems like a workaround not fixing the real issue.  I think we could have incorrect results if `root_globals[name]` and `f_globals[name]` point to different objects.  Which is possible...

Where is the `NameError` coming from?

**Comment by anijain2305** [comment]
@jansel The repro is described here - https://github.com/pytorch/torchdynamo/issues/1126


The forward is in this file - https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/speech_transformer/speech_transformer/transformer/encoder.py

If you look at the imports at the top, it does not have `import torch`.

The analysis of `forward` inlines functions in these file - https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/speech_transformer/speech_transformer/transformer/attention.py

The NameError is coming after the inlining is done at runtime.

**Review by jansel — CHANGES_REQUESTED** [review]
I think I found the root cause for this, let me confirm.

**Comment by jansel** [comment]
Root cause fix is here: #1129

File: torchdynamo/symbolic_convert.py [hunk 2]
```diff
@@ -399,6 +399,8 @@ def get_global_source(self, name):
                 source = AttrSource(
                     self.import_source(self.f_globals["__name__"]), name
                 )
+                if name not in self.output.root_globals and name in self.f_globals:
```
Comment (jansel) [inline torchdynamo/symbolic_convert.py:402]: Funny enough, I think this fix only worked because some completely unrelated code we inline used the global `torch`.  The reference to `torch` was coming from unspecialized integer/float graph inputs and had nothing to do with user code or inlining.

If you modify the inlined function to not use `torch.*` this fix will stop working.

See #1129

**Comment by anijain2305** [comment]
Closed in favor of #1129
